File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hp = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in hp:
                return [hp[diff], i]
            hp[nums[i]] = i


        

File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> res;
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<int> output;
        int sum = 0;
        backtracking(nums, target, output, sum, 0);
        return res;
    }

    void backtracking(vector<int> &nums, int target, vector<int> &output, int sum, int idx) {
        if (sum == target) {
            res.push_back(output);
        }
        if (sum > target) return;

        for (int i = idx; i < nums.size(); i++) {
            output.push_back(nums[i]);
            backtracking(nums, target, output, sum + nums[i], i);
            output.pop_back();
        }
    }
};
